sbi/inference/snpe/snpe_base.py

```diff
@@ -391,26 +386,21 @@ def _train(
 
             # Calculate validation performance.
             self._neural_posterior.neural_net.eval()
-            log_prob_sum = 0
+            batch_loss_sum = 0
             with torch.no_grad():
                 for batch in val_loader:
                     theta_batch, x_batch, masks = (
                         batch[0].to(self._device),
                         batch[1].to(self._device),
                         batch[2].to(self._device),
                     )
-                    # Just do maximum likelihood in the first round.
-                    if round_ == 0:
-                        log_prob = self._neural_posterior.neural_net.log_prob(
-                            theta_batch, x_batch
-                        )
-                    else:
-                        log_prob = self._get_log_prob_proposal_posterior(
-                            theta_batch, x_batch, masks
-                        )
-                    log_prob_sum += log_prob.sum().item()
-
-            self._val_log_prob = log_prob_sum / num_validation_examples
+                    # Calculate loss as calibration kernel times log prob.
```

Comment:
The docstring says it? If we add a function but we also keep the comment we may end up being over ceremonious (i.e. maybe inline the function, remove the comment, rename the function). One has to consider what to do in this decision space.